tests/integration/test_simulation.py

```diff
@@ -14,112 +14,416 @@
+    })
+    schd = scheduler(id_)
+    async with start(schd):
+        # Submit first psuedo-job and "run" to failure:
+        one_1066 = schd.pool.get_tasks()[0]
+
+        itask = run_simjob(schd, one_1066.point, 'one')
+        assert ['failed', 'failed', False] in itask.state.outputs.get_all()
+
+        # Modify config as if reinstall had taken place:
+        conf_file = Path(schd.workflow_run_dir) / 'flow.cylc'
+        conf_file.write_text(
+            conf_file.read_text().replace('False', 'True'))
+
+        # Reload Workflow:
+        await schd.command_reload_workflow()
+
+        # Submit second psuedo-job and "run" to success:
+        itask = run_simjob(schd, one_1066.point, 'one')
+        assert [
+            'succeeded', 'succeeded', True] in itask.state.outputs.get_all()
+
+
+async def test_settings_broadcast(
+    flow, scheduler, start, monkeytime
+):
+    """Assert that broadcasting a change in the settings for a task
+    affects subsequent psuedo-submissions.
+    """
+    id_ = flow({
+        'scheduler': {'cycle point format': '%Y'},
+        'scheduling': {
+            'initial cycle point': '1066',
+            'graph': {'R1': 'one'}
+        },
+        'runtime': {
+            'one': {
+                'execution time limit': 'PT1S',
+                'simulation': {
+                    'speedup factor': 1,
+                    'fail cycle points': '1066',
+                }
+            },
+        }
+    }, defaults=False)
+    schd = scheduler(id_, paused_start=False, run_mode='simulation')
+    async with start(schd):
+        itask = schd.pool.get_tasks()[0]
+        itask.state.is_queued = False
+
+        # Submit the first - the sim task will fail:
+        schd.task_job_mgr._simulation_submit_task_jobs(
+            [itask], schd.workflow)
+        assert itask.mode_settings.sim_task_fails is True
+
+        # Let task finish.
+        monkeytime(itask.mode_settings.timeout + 1)
+        assert sim_time_check(
+            schd.task_events_mgr, [itask],
+            schd.workflow_db_mgr
+        ) is True
+
+        # The mode_settings object has been cleared:
+        assert itask.mode_settings is None
+
+        # Change a setting using broadcast:
+        schd.task_events_mgr.broadcast_mgr.put_broadcast(
+            ['1066'], ['one'], [{
+                'simulation': {'fail cycle points': ''}
+            }])
+
+        # Assert that list of broadcasts doesn't change if we submit
+        # Invalid fail cycle points to broadcast.
+        schd.task_events_mgr.broadcast_mgr.put_broadcast(
+            ['1066'], ['one'], [{
+                'simulation': {'fail cycle points': 'higadfuhasgiurguj'}
+            }])
+        schd.task_events_mgr.broadcast_mgr.put_broadcast(
+            ['1066'], ['one'], [{
+                'simulation': {'fail cycle points': '1'}
+            }])
+        assert {'fail cycle points': []} == (
+            schd.broadcast_mgr.broadcasts['1066']['one']['simulation'])
```

Comment:
Yoda expression
```suggestion
        assert schd.broadcast_mgr.broadcasts['1066']['one'][
            'simulation'
        ] == {'fail cycle points': []}
```

Reply:
> Yoda expression

And I suddenly understood what you meant!